assets/src/dashboard/app/views/myStories/karma/myStories.karma.js

```diff
@@ -436,6 +504,182 @@ describe('Grid view', () => {
       expect(renderedStoriesById).toEqual(storiesOrderById);
     });
   });
+
+  describe('Dashboard keyboard navigation and focus logic', () => {
+    it('should navigate the grid via keyboard', async () => {
+      const storyCards = fixture.screen.getAllByTestId(/card-action-container/);
+      await focusOnGridByKeyboard();
+
+      for (let i = 0; i < storyCards.length; i++) {
+        const expectedStory = storyCards[i];
+        // eslint-disable-next-line no-await-in-loop
+        await fixture.events.keyboard.press('right');
+        expect(expectedStory).toEqual(document.activeElement);
+      }
+    });
+
+    it('should focus on context menu items via keyboard', async () => {
+      const storyCards = fixture.screen.getAllByTestId(/card-action-container/);
+      const selectedStory = storyCards[0];
+      await focusOnGridByKeyboard();
+      await fixture.events.keyboard.press('right');
+
+      // we should have focused the first story in the grid
+      expect(selectedStory).toEqual(document.activeElement);
+
+      let limit = 0;
+      const [contextMenuKabab] = fixture.screen.getAllByLabelText(
+        'More Options'
+      );
+
+      // tab to the button which opens the context menu
+      while (!contextMenuKabab.contains(document.activeElement) && limit < 4) {
+        // eslint-disable-next-line no-await-in-loop
+        await fixture.events.keyboard.press('tab');
+        limit++;
+      }
+
+      expect(contextMenuKabab).toEqual(contextMenuKabab);
+
+      await fixture.events.keyboard.press('Enter');
+
+      // now the focused item should be the first context menu item
+      const [contextMenuList] = fixture.screen.getAllByTestId(
+        /context-menu-list/
+      );
```

Comment:
💭 If we add `aria-expanded={isOpen` to the menu, we could find this dialog with `get` instead of getting all of them since it will be the only open dialog. Would that be easier here?

Reply:
`getBy` throws an error if it finds multiple elements with the test id whether the other properties match. Since there are multiple lists, I have to use the `getAllBy`, but I added the aria attribute anyway as this still makes the test more robust.